gateway/ln-gateway/src/actor.rs

```diff
@@ -229,6 +229,33 @@ impl GatewayActor {
             .map(|out_point| out_point.txid)
     }
 
+    pub async fn backup(&self) -> Result<()> {
```

Comment:
should we be looking to call backup periodically, or even best-effort as the gateway is looking to shut down?
why does the client need to initiate backup, and not have that happen automatically?

Reply:
We could. We never wired it anywhere.

Reply:
Now I'm a bit worried about making backup of corrupted state. :D

Reply:
Yea. this is a nice change, but I don't think it will fix your problem. Your backup will still contain invalid `NotIssuanceRequest`s which will prevent your valid `NotIssuanceRequest`s from completing.

I think fixing `https://github.com/fedimint/fedimint/issues/1479` would solve the problem. You'd still have invalid `NotIssuanceRequest`s in your DB, but they would just cause a little delay when fetching balance.

Reply:
Actually I think this change could work on its own, but only if you never made a backup to the federation and restored by downloading entire epoch history. Then it wouldn't accidentally pick up the invalid `NotIssuanceRequest`s.

Reply:
https://github.com/fedimint/fedimint/blob/73b79347e370594794a0500cbb39dac9468c5756/client/client-lib/src/mint/backup.rs#L114

We do back-up, track, wipe old & restore from backup https://github.com/fedimint/fedimint/blob/73b79347e370594794a0500cbb39dac9468c5756/client/client-lib/src/mint/backup.rs#L114 `NoteIssuanceRequest`s

Reply:
Nevermind, I think this will work!